cmake-format.yaml

```diff
@@ -0,0 +1,13 @@
+# https://github.com/cheshirekow/cmake_format
```

Comment:
I think it is better to leave this to a separate PR. And there are additional work to make this tool usable for everyone:
- install cmake-format in `./script/bootsrap.sh`
- check CMakeLists.txt files in `./script/check-pretty.sh`
- format CMakeLists.txt files in `./script/make-pretty.sh`
- for the actual cmake style, we need to discuss with more openthread contributors to agree on it.

@gocarlos What do you think?

Reply:
I'm ok with that